src/main/java/pl/sda/eventmanager/services/RegisterValidator.java

```diff
@@ -0,0 +1,53 @@
+package pl.sda.eventmanager.services;
+
+import org.springframework.stereotype.Service;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+import pl.sda.eventmanager.dto.RegisterForm;
+
+@Service
+public class RegisterValidator implements Validator {
+
+    private final UserService userService;
+
+    public RegisterValidator(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public boolean supports(Class<?> aClass) {
+        return RegisterForm.class.equals(aClass);
+    }
+
+    @Override
+    public void validate(Object o, Errors errors) {
+        RegisterForm registerForm = (RegisterForm) o;
+
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "email", "NotEmpty", "All fields are required.");
+        if (userService.findByEmail(registerForm.getEmail()).isPresent()) {
```

Comment:
currently you're performing 2 SQL queries to check email and nickname uniqueness, this can be reduced to 1, more f2f.